### PR TITLE
[SF.10] Simplify example

### PR DESCRIPTION
**Pull request by rico-chet**
The original example uses a std header to demonstrate transitive
inclusion, which makes it relate to real code, on the one hand. Yet, the
header is too long to be put in the example, so it's omitted and thus
the example is rather obscure, on the other hand.

Switch to a constructed example which uses a short header which is
presented right there to make the point clearer.

### CODE REVIEW

**Comment by cubbimew** [comment]
The original example demonstrates a common beginner blunder, how is it "obscure"?

**Comment by hsutter** [comment]
Editors call: We prefer the original example because it's a real example and a little more concise.

**Comment by rico-chet** [comment]
> The original example demonstrates a common beginner blunder, how is it "obscure"?

It's obscure as readers need to look up where `getline()` is supposed to be declared (e.g. I don't know by heart because I don't use the function much). To me, the real issue with violating the rule is that code gets fragile and prone to breaking on refactoring of included headers. Compile-time errors at the time offending code (i.e. missing an `#include`) is written isn't a big issue, IMHO.

So in short, I believe currently the example fails to demonstrate what the rule is good for.

If you aim for examples which are rather concise and less explanatory (require readers to do research), I don't mind.
